LoopUI/Views/CGMStatusHUDView.swift

```diff
@@ -0,0 +1,172 @@
+//
+//  CGMStatusHUDView.swift
+//  LoopUI
+//
+//  Created by Nathaniel Hamming on 2020-06-05.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import UIKit
+import HealthKit
+import LoopKit
+import LoopKitUI
+
+public final class CGMStatusHUDView: DeviceStatusHUDView, NibLoadable {
+    
+    @IBOutlet public weak var glucoseValueHUD: GlucoseValueHUDView!
+    
+    @IBOutlet public weak var glucoseTrendHUD: GlucoseTrendHUDView!
+    
+    override public var orderPriority: HUDViewOrderPriority {
+        return 1
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+    
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+        
+    }
+    
+    override func setup() {
+        super.setup()
+        alertStatusView.setIconPosition(.right)
+    }
+    
+    public override func tintColorDidChange() {
+        super.tintColorDidChange()
+        
+        glucoseValueHUD.tintColor = tintColor
+        glucoseTrendHUD.tintColor = tintColor
+    }
+    
+    public func presentAddCGMAlert() {
+        alertStatusView.alertMessageLabel.text = LocalizedString("Add CGM", comment: "Title text for button to set up a CGM")
+        alertStatusView.alertMessageLabel.tintColor = .label
+        alertStatusView.alertIcon.image = UIImage(systemName: "plus.circle")
+        alertStatusView.alertIcon.tintColor = .systemBlue
+        presentAlert()
+    }
+    
+    override public func presentAlert() {
+        guard !statusStackView.arrangedSubviews.contains(alertStatusView) else {
+            return
+        }
+        
+        // need to also hide these view, since they will be added back to the stack at some point
+        glucoseValueHUD.isHidden = true
+        glucoseTrendHUD.isHidden = true
+        statusStackView.removeArrangedSubview(glucoseValueHUD)
+        statusStackView.removeArrangedSubview(glucoseTrendHUD)
+        
+        super.presentAlert()
+    }
+    
+    override public func dismissAlert() {
+        guard statusStackView.arrangedSubviews.contains(alertStatusView) else {
+            return
+        }
+        
+        super.dismissAlert()
+        
+        statusStackView.addArrangedSubview(glucoseValueHUD)
+        statusStackView.addArrangedSubview(glucoseTrendHUD)
+        glucoseValueHUD.isHidden = false
+        glucoseTrendHUD.isHidden = false
+    }
+    
+    public func setGlucoseQuantity(_ glucoseQuantity: Double, at glucoseStartDate: Date, unit: HKUnit, staleGlucoseAge: TimeInterval, sensor: SensorDisplayable?) {
+        var accessibilityStrings = [String]()
```

Comment:
This feels like it should be a function in a `ViewModel`, and put under unit test.

Reply:
I'll include a comment and handle this with the LOOP-1293, which is the CGM specific ticket for the status bar.